users.js

```diff
@@ -1375,6 +1379,7 @@ class User {
 
 		let stillInRoom = false;
 		if (connection) {
+			// @ts-ignore TypeScript inferring wrong type for room
```

Comment:
tbh in general all this is caused by, for some reason union types aren't supported by `global.d.ts` for some reason.

Reply:
ohhhhh wait I know why now

Reply:
I DID IT, IT ONLY TOOK A FUCKTON OF TIME BUT I REALIZED WHAT WAS WRONG